### PR TITLE
Improve PHPDoc for package requires

### PR DESCRIPTION
**Pull request by Soullivaneuh**

### CODE REVIEW

**Comment by alcohol** [comment]
Please use `/** Link[] */` for the class properties.

**Comment by Soullivaneuh** [comment]
> Please use `/** Link[] */` for the class properties.

I'm sorry but I don't think this is the right convention for this.

BTW, this will be not managed by cs-fixer tools (if you use it) AFAIK.

Do you have a link about that? I'm curious. :-)

**Comment by alcohol** [comment]
It's fine.

**Comment by staabm** [comment]
@Soullivaneuh I guess he meant write it on a single line and forgot the `@var`

**Comment by alcohol** [comment]
I did not forget `@var`... :trollface:

**Comment by staabm** [comment]
:-)

**Comment by Soullivaneuh** [comment]
> I guess he meant write it on a single line and forgot the `@var`

No, bad copy paste apparently. :-)

> It's fine.

Fine what? :-)

**Comment by alcohol** [comment]
It's fine to put it on one line.

File: src/Composer/Package/Package.php [hunk 1]
```diff
@@ -43,10 +43,17 @@ class Package extends BasePackage
     protected $stability;
     protected $notificationUrl;
 
+    /**
+     * @var Link[]
+     */
```
Comment (stof) [inline src/Composer/Package/Package.php]: this should also be done for other properties (conflicts, provides and replaces are also package links)

Comment (Soullivaneuh) [inline src/Composer/Package/Package.php]: This PR was for requirement properties. But yeah I can do that.

@stof In the same time, could I please have your opinion about @alcohol comment? https://github.com/composer/composer/pull/4508#issuecomment-149172694

Comment (stof) [inline src/Composer/Package/Package.php]: Well, I don't really care whether it is single line or multiline here? It is a matter of preferences IMO

Comment (alcohol) [inline src/Composer/Package/Package.php]: I personally find it a bit ridiculous to use 30 lines of comments to document 10 lines of class properties, when you can achieve the same effect using just 10 lines.

**Comment by Soullivaneuh** [comment]
Where are you with this PR? Should I do another things? Document each property as @stof asked?

**Comment by alcohol** [comment]
If you could do that, as one-liners, I'll merge it for you, sure.

**Comment by Soullivaneuh** [comment]
Will do this on a new PR. Lost the repository of this one.

**Comment by Soullivaneuh** [comment]
@alcohol Please see #4676.

**Pull request closed without merging**